pkg/api/registry/registry.proto

```diff
@@ -34,25 +34,19 @@ message NetworkServiceEndpoint {
     google.protobuf.Timestamp expiration_time = 5;
 }
 
-service NetworkServiceEndpointRegistry {
-    rpc Register (NetworkServiceEndpoint) returns (NetworkServiceEndpoint);
-    rpc Find (NetworkServiceEndpointQuery) returns (stream NetworkServiceEndpoint);
-    rpc Unregister (NetworkServiceEndpoint) returns (google.protobuf.Empty);
-}
-
 service NetworkServiceRegistry {
-    rpc Register (NetworkService) returns (NetworkService);
-    rpc Find (NetworkServiceQuery) returns (stream NetworkService);
-    rpc Unregister (NetworkService) returns (google.protobuf.Empty);
-}
-
-message NetworkServiceQuery {
-    NetworkService network_service = 1;
-    bool watch = 2;
+    rpc Register (Record) returns (Record);
```

Comment:
> Should every NSE also register own Network Service?

Yes, every NSE should register NS. Also as it implemented in mono repo https://github.com/networkservicemesh/networkservicemesh/blob/fd01d583f32ef9c143e43460f54910793d551964/sdk/endpoint/endpoint.go#L114

> How they will be merged in case multiple values exist

Register() will just update exist NS. In case when we deploy the same NSEs it should be ok. For the case when the user deployed NSE with NS which already was created by other NSE we can add for SDK some function like `checkNS()` that will return an error if NS with the same name and **different fields** already exists.

> How they will be removed in case of Unregister

NS can be removed only if no one NSE doesn't need this NS. Also, NS can be removed by the registry if we have not endpoints for this NS.